lib/games_manager.rb

```diff
@@ -171,81 +139,68 @@ def get_season_games(season)
       game.game_id
     end
   end
-  
+
   def total_home_goals
     team_id_by_home_goals = games.map do |game|
       [game.home_team_id, game.home_goals]
     end
     sum_values(team_id_by_home_goals)
   end
-  
+
   def total_away_goals
     team_id_by_away_goals = games.map do |game|
       [game.away_team_id, game.away_goals]
     end
     sum_values(team_id_by_away_goals)
   end
-  
+
   def total_home_games
     team_id_by_home_games = games.map do |game|
       [game.home_team_id, 1]
     end
     sum_values(team_id_by_home_games)
   end
-  
+
   def total_away_games
     team_id_by_away_games = games.map do |game|
       [game.away_team_id, 1]
     end
     sum_values(team_id_by_away_games)
   end
-  
+
   def sum_values(key_value_arr)
     sum = Hash.new(0)
     key_value_arr.each do |key, value|
       sum[key] += value
     end
     sum
   end
-  
+
   def highest_scoring_visitor
     averages = total_away_goals.merge(total_away_games) do |team_id, goals, games|
       (goals/games.to_f).round(2)
     end
-    average_max = averages.max_by do |team_id, average|
-      average
-    end
-    average_max[0]
+    (averages.max_by {|team_id, average| average})[0]
```

Comment:
This refactor and my refactor of the same methods would complement each other nicely!